File: web/src/services/userService.ts
import { api } from './api'
import type { User, UserCreate } from '../models/user'

export const userService = {
  async list(): Promise<User[]> { return (await api.get<User[]>('/users')).data },
  async create(payload: UserCreate): Promise<User> { return (await api.post<User>('/users', payload)).data },
}

File: backend/app/services/user_service.py
from sqlalchemy.orm import Session

from app.repositories.user_repository import UserRepository
from app.schemas.user import UserCreate


class UserService:
    def __init__(self, db: Session):
        self.repository = UserRepository(db)

    def list_users(self):
        return self.repository.list()

    def create_user(self, data: UserCreate):
        return self.repository.create(data.email, data.full_name)

